File: app/src/main/java/com/example/madarsofttask/data/db/MyDAO.java
package com.example.madarsofttask.data.db;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import com.example.madarsofttask.data.models.User;
import java.util.List;

@Dao
public interface MyDAO {

    @Insert
    void addUser(User user);

    @Query("select * from users")
    List<User> retrieveUsers();

}

File: app/src/main/java/com/example/madarsofttask/data/db/MyDAO.kt
package com.example.madarsofttask.data.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.example.madarsofttask.data.models.User

@Dao
interface MyDAO {
    @Insert
    fun addUser(user: User?)

    @Query("select * from users")
    fun retrieveUsers(): LiveData<List<User?>?>
}